feat: added invoices page

PavelDenchev opened the pull request:
Added button in sidebar that leads to the Invoices page.
Invoices page has functionality to view, add, edit and delete Invoices.

Video link: https://www.loom.com/share/a2105a4aca314e8d855231e31ef8a8fb

evereq commented:
@rmagon merging it, please review but looks very good already for me 🚀 

Following changes was requested to do in UI:
-  put "Total Value" on a separate line
- make "Terms" to be multi-line input (terms should go above "Paid" checkbox)
- We should auto-generate (on the server) Invoice Number, but allow a customer to edit it if needed, 
i.e. don't use generate in DB, but use some logic like: "Load last invoice from Db, see the number, make +1 and return". So if we have invoice 53, the new number will be filled with 54, 
but customer can make it 66 (make it as a separate function on server to get new invoice number)
- In the Grid, where we display Value, needs to display Currency too

@PavelDenchev see above list of requested changes, you can continue pushing to same branch!

PavelDenchev commented:
Changed layout of Add/Edit forms. Added currency to Total Value in table. Invoice Number now auto generates.

Video: https://www.loom.com/share/530a1cdd29c240a2bab04a53ec5363f9

rmagon commented:
Looks good! @PavelDenchev I see it's marked as a draft PR but I guess it's ready to merge?

rmagon commented on apps/api/src/app/invoice-item/invoice-item.entity.ts:
```diff
@@ -0,0 +1,48 @@
+	@Column()
+	invoiceId: string;
+
+	@ApiPropertyOptional({ type: Invoice, isArray: true })
```
@PavelDenchev this one :)
